### DISCUSSION TITLE
Bot shutdown not executing

### DISCUSSION THREAD

**Discussion by rozkomnadzor** [aiogram/aiogram#1326] [Q&A]
I launch the bot and everything works stably, the "startup" event is processed, but when the bot is terminated, the shutdown event is ignored. How can I terminate it correctly so that it works correctly?

```
if __name__ == '__main__':
    loop = asyncio.new_event_loop()
    asyncio.set_event_loop(loop)
    try:
        loop.run_until_complete(start_bot())
    except KeyboardInterrupt:
        pass
```

```
async def start_bot():
    # await init_beanie(database=mongo_client.db, document_models=[LogMessage]) 
    await dp.start_polling(bot)
```

```
async def shutdown_event(bot: Bot, dispatcher: Dispatcher):
    await dispatcher.storage.close()
    await bot.session.close()
    print('stopped')

async def startup_event():
    print('started')
```

**Comment by rozkomnadzor** [comment]
I found that the problem is in `loop.run_until_complete` but without this method I get the error `Event loop is closed` upon completion

**Reply by Olegt0rr** [reply]
Use only one loop per app with single entrypoint

```python
async def main():
    """Launch everything!"""

asyncio.run(main())
```

**Reply by OGURETS13** [reply]
yes, this is how i tried it as well, but the point is - aiogram seems to stop only it's own polling, but not the other tasks

**Comment by OGURETS13** [comment]
dp.start_polling(bot, handle_signals=False)

this way you can stop aiogram handling your interrupt signals
